_posts/2015-07-21-CocoaPods-0.38.markdown

```diff
@@ -0,0 +1,152 @@
+| `sandbox`                 | `sandbox`                                   |
+| `config`                  | `config`                                    |
+| `installer`               | not needed                                  |
+
+See also the [gem API documentation of the `Installer`](http://www.rubydoc.info/github/cocoapods/cocoapods/Pod/Installer).
+
+If you need to keep compatibility to the older versions of CocoaPods, then you can simply check the gem version in the Podfile as it is just Ruby code after all.
+
+```ruby
+post_install do |installer_or_rep|
+  # @note: Remove after CocoaPods 0.38+ is completely rolled out
+  # and rename block argument to `installer`.
+  installer = installer_or_rep.respond_to?(:installer) ? installer_or_rep.installer : installer_or_rep
+  
+  # You can use the installer from now on.
+end
+```
+
+
+### Split of `xcconfig`
+
+One of the more advanced attribute in Podspecs is - _or better was_ - the attribute `xcconfig`.
+
+This allow you to specify build settings for your Pod.
+Previously those were made available to libraries and the integrated target.
+
+We broke this with the support for frameworks.
+This wasn't an issue for most Pods because the build setting is mostly used to configure build settings which are relevant only to the generated Pod target itself.
+When we mentioned that there were still some Pods which weren't working anymore as expected due to that change, we decided to split the attribute from now on.
+This way Pods also can't accidentally overwrite or conflict with build settings of the user target.
+
+The former attribute `xcconfig` is deprecated and will cause a linter error when pushing new versions to trunk.
+The new attributes are available as `pod_target_xcconfig` and `user_target_xcconfig`, which makes their effects more clear.
+The latter attribute (`user_target_xcconfig`) should be used with great care, because well designed Pods should be self-contained and make as few assumptions about their environment as possible.
+Furthermore, this attribute can cause conflicts when different values are specified by two Pods for a build setting which doesn't allow multiple values and so cannot be merged.
+
+
+### Deterministic UUIDs by Xcodeproj
+
+This update ships also the latest version of Xcodeproj, which encapsulates the API to Xcode projects.
+Beside support for new build settings, it now supports to generate deterministic UUIDs for file references.
+That means in practice mostly, that you have smaller diffs for the `Pods.xcodeproj` after running `pod install` or `pod update`, which reduce unnecessary merge conflicts.
+This behavior can be disabled via the new `COCOAPODS_DISABLE_DETERMINISTIC_UUIDS` environment variable.
+
+
+### Resolver takes the Deployment Target into account
+
+The deployment target of a podspec is now taken into account in the dependency resolution process.
+This will prevent errors, which can appear, when maintaining podspecs for legacy versions of the toolchain.
```

Comment:
`This will ensure that CocoaPods will always be able to resolve your Podfile, even if newer versions of pods are released that have higher minimum deployment targets.`